app/code/Magento/Sales/Block/Order/Recent.php

```diff
@@ -28,23 +39,31 @@ class Recent extends \Magento\Framework\View\Element\Template
      */
     protected $_orderConfig;
 
+    /**
+     * @var \Magento\Store\Model\StoreManagerInterface
+     */
+    private $storeManager;
+
     /**
      * @param \Magento\Framework\View\Element\Template\Context $context
      * @param \Magento\Sales\Model\ResourceModel\Order\CollectionFactory $orderCollectionFactory
      * @param \Magento\Customer\Model\Session $customerSession
      * @param \Magento\Sales\Model\Order\Config $orderConfig
+     * @param \Magento\Store\Model\StoreManagerInterface $storeManager
      * @param array $data
      */
     public function __construct(
-        \Magento\Framework\View\Element\Template\Context $context,
-        \Magento\Sales\Model\ResourceModel\Order\CollectionFactory $orderCollectionFactory,
-        \Magento\Customer\Model\Session $customerSession,
-        \Magento\Sales\Model\Order\Config $orderConfig,
+        Context $context,
+        CollectionFactory $orderCollectionFactory,
+        Session $customerSession,
+        Config $orderConfig,
+        StoreManagerInterface $storeManager,
```

Comment:
Hi @Coderimus, did you have a look to retrieve the current store id other than via adding a new dependency with StoreManager?

Reply:
If we need this new dependency it needs to be backwards compatible, see here: http://devdocs.magento.com/guides/v2.0/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter

Reply:
@avoelkl thank you so much for your response and review! 
I think I can do this without including new dependency using `$this->_storeManager` property in the Block class and leave the unit test with the Mock object of the storeManagerInterface. As I can see they are the same objects.
I will implement this change, test it and deploy to review ASAP.
Thank you,
Alex

Reply:
Hi @Coderimus, thanks for your reply. Let me know if you need some help!